### PR TITLE
Work

### PR DESCRIPTION
**Pull request by fearthecowboy**
- Complete initial schema parsing
- Added:
  - created a host interface for the deserializer and added events 
  - tracking perf during tests
  - remove the attic file so we can tell if it's writing it.
  - moved files so that there can common code between oai2 and oai3 deserialization
  - refactored anonymous names a bit
Fixed
  - fix bug on addUnusedTo during arrays

### CODE REVIEW

File: adl/core/serialization/openapi/v3/schema.ts [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+  const combineWith = new Array<Schema>();
+  if (objectSchema) {
+    combineWith.push(objectSchema);
+  }
+  if (oneOf) {
+    combineWith.push(oneOf);
+  }
+  // if (allOf) {
+  // combineWith.push(allOf);
+  // }
```
Comment (nguerrera) [inline adl/core/serialization/openapi/v3/schema.ts:163]: Delete?

File: adl/core/serialization/openapi/v3/schema.ts [hunk 2]
```diff
@@ -0,0 +1,746 @@
...
+export async function processDateSchema(schema: v3.Schema, $: Context): Promise<Schema | undefined> {
+  if ($.forbiddenProperties(schema, ...stringProperties, ...objectProperties, ...arrayProperties, ...numberProperties)) {
+    return undefined;
+  }
+
+  return $.api.schemas.Date;
+}
+
+export async function processTimeSchema(schema: v3.Schema, $: Context): Promise<Schema | undefined> {
+  if ($.forbiddenProperties(schema, ...stringProperties, ...objectProperties, ...arrayProperties, ...numberProperties)) {
```
Comment (nguerrera) [inline adl/core/serialization/openapi/v3/schema.ts:426]: Extract helper for this pattern?

File: adl/core/model/schema.ts [hunk 3]
```diff
@@ -24,6 +34,28 @@ export class Schema extends Element {
...
+   * A property MUST NOT be marked as both readOnly and writeOnly being true. 
+   * Default value is false.
+   * 
+   */
+  writeonly?: boolean;
+
+  /**
+  * description of the property
+  * CommonMark syntax MAY be used for rich text representation.
+  */
```
Comment (nguerrera) [inline adl/core/model/schema.ts:56]: nit: misaligned asterisks

File: adl/core/model/schema.ts [hunk 4]
```diff
@@ -1,12 +1,21 @@
...
 
 export class Schema extends Element {
   anonymous?: boolean;
+  /** name of this schema - may be an alias or an actual string name */
+  name: SchemaName;
 
-  /* short description */
+  /**
+   * short description of the schema
+   *
```
Comment (nguerrera) [inline adl/core/model/schema.ts:11]: nit: single line like name above?

File: adl/core/serialization/openapiv3/schema.ts [hunk 5]
```diff
@@ -46,17 +61,229 @@ export async function processSchemaReference(ref: JsonReference<v3.Schema>, $: C
...
+  }
+
+  // no object combinations
+  const result = new XorSchema(schemaName, schemas, commonProperties(schema));
+  $.api.schemas.combinations.push(result);
+  return result;
+}
+
+export async function processSillyRef(schema: v3.Schema, $: Context, options?: { isAnonymous?: boolean; forUnderlyingEnumType?: boolean }): Promise<Schema | undefined> {
+  throw undefined;
```
Comment (nguerrera) [inline adl/core/serialization/openapiv3/schema.ts]: throw an Error with a description here?

File: adl/core/serialization/openapi/v3/schema.ts [hunk 6]
```diff
@@ -0,0 +1,746 @@
...
+  }
+
+  // no object combinations
+  const result = new XorSchema(schemaName, schemas, commonProperties(schema));
+  $.api.schemas.combinations.push(result);
+  return result;
+}
+
+export async function processSillyRef(schema: v3.Schema, $: Context, options?: { isAnonymous?: boolean; forUnderlyingEnumType?: boolean }): Promise<Schema | undefined> {
+  throw undefined;
```
Comment (nguerrera) [inline adl/core/serialization/openapi/v3/schema.ts:224]: Throw an error with a description here?

File: adl/core/model/schema.ts (hunk 3 above)
Comment (fearthecowboy) [reply adl/core/model/schema.ts]: lol

Comment (nguerrera) [reply adl/core/model/schema.ts]: (sorry)

**Pull request merged**